backends/arm/test/runner_utils.py

```diff
@@ -0,0 +1,492 @@
+        ), f"TOSA artifact path don't exist! Path: {self.intermediate_path}"
+
+        self.is_quantized: bool = False
+        self.input_names: list[str] = None
+        self.output_name: str = None
+        self.qp_input: list[QuantizationParams] = None
+        self.qp_output: QuantizationParams = None
+        self.timeout = 120
+
+        self._has_init_run = False
+
+    def init_run(self, exported_program: ExportedProgram, is_quantized: bool):
+        self.input_names = _get_input_names(exported_program)
+        self.output_node = _get_output_node(exported_program)
+        self.output_name = self.output_node.name
+        self.is_quantized = is_quantized
+
+        if is_quantized:
+            self.qp_input = _get_input_quantization_params(
+                exported_program, self.input_names
+            )
+            self.qp_output = _get_output_quantization_params(
+                exported_program, self.output_node
+            )
+        else:
+            self.qp_input = [None] * len(self.input_names)
+            self.qp_output = None
+
+        self._has_init_run = True
+
+    def set_timeout(self, timeout: int):
+        self.timeout = timeout
+
+    def run_corstone300(
+        self,
+        inputs: Tuple[torch.Tensor],
+    ) -> torch.Tensor:
+
+        assert (
+            self._has_init_run
+        ), "RunnerUtil needs to be initialized using init_run() before running Corstone300."
+
+        pte_path = os.path.join(self.intermediate_path, "program.pte")
+        assert os.path.exists(pte_path), f"Pte path '{pte_path}' not found."
+
+        for input_name, quant_param, data in zip(
+            self.input_names, self.qp_input, inputs
+        ):
+            save_bytes(self.intermediate_path, data, False, input_name, quant_param)
+
+        out_path = os.path.join(self.intermediate_path, "out")
+        out_path_with_suffix = out_path + "-0.bin"
+        input_paths = []
+        for name in self.input_names:
+            input_paths.append(
+                os.path.join(self.intermediate_path, f"{name}.bin"),
+            )
+        elf_path = os.path.join(
+            "cmake-out", "arm_semihosting_executor_runner", "arm_executor_runner"
+        )
+        assert os.path.exists(
+            elf_path
+        ), f"Did not find build arm_executor_runner in path {elf_path}, run setup_testing.sh?"
+
+        cmd_line = f"executor_runner -m {pte_path} -o {out_path}"
+        for input_path in input_paths:
+            cmd_line += f" -i {input_path}"
+
+        command_args = [
+            "FVP_Corstone_SSE-300_Ethos-U55",
+            "-C",
+            "ethosu.num_macs=128",
+            "-C",
+            "mps3_board.visualisation.disable-visualisation=1",
+            "-C",
+            "mps3_board.telnetterminal0.start_telnet=0",
+            "-C",
+            "mps3_board.uart0.out_file='-'",
+            "-C",
+            "cpu0.CFGITCMSZ=11",
+            "-C",
+            "cpu0.semihosting-enable=1",
```

Comment:
Why do we want to do this as opposed to loading on a "real" TCM or something? Curious from test coverage point of view. I understand if there are issues with address space etc.

Reply:
Are you referring to the increased ITCM size? This was somewhat of a quick fix to fit the runner in the FVP in a simple way. But It would be nice to match the Himax or similar in the future though.

Reply:
I am referring to using SemiHosting for testing

Reply:
Ok, the reason for that is to avoid building a new runner for every test